pydantic/_internal/_typing_extra.py

```diff
@@ -293,7 +293,7 @@ def get_function_type_hints(
     except AttributeError:
         type_hints = get_type_hints(function)
         if isinstance(function, type):
-            type_hints.setdefault('return', type)
+            type_hints.setdefault('return', function)
```

Comment:
@sydney-runkle, I believe you mean to use the function (well in this case the type) here in your previous fix?

Reply:
Great, thanks!